helion/_compiler/tile_strategy.py

```diff
@@ -516,3 +516,19 @@ def combine(self, other: CompactedShape) -> CompactedShape:
             user_indices=[*self.user_indices, *other.user_indices],
             block_indices=[*self.block_indices, *other.block_indices],
         )
+
+
+class NDGridTileStrategy(NDTileStrategy):
+    def __init__(
+        self,
+        fn: DeviceFunction,
+        block_indices: list[int],
+        loop_order: list[int],
+    ) -> None:
+        super().__init__(
+            fn=fn,
+            block_indices=block_indices,
+            block_size=[1] * len(block_indices),  # pyre-ignore[6]
+            loop_order=loop_order,
+            l2_grouping=1,
+        )
```

Comment:
I made `NDGridTileStrategy` subclass from `NDTileStrategy` because they share most of the `codegen_grid()` code. But one difference is that `l2_grouping` is (iiuc) not needed for grid indexing so I explicitly set it to 0 here.

If we don't want it to subclass from `NDTileStrategy`, I can also move the shared logic of `NDTileStrategy.codegen_grid()` into a common function called by both `NDGridTileStrategy` and `NDTileStrategy`.

Reply:
Refactoring into a common base class would be useful.